File: Data Structures & Algorithms/anagram-groups/submission-2.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        let count = {};
        let res = [];
        for(let i = 0; i < strs.length; i++) {
            const key = strs[i].split("").sort().join("");
            if(!count[key]) {
                count[key] = [strs[i]];
            } else {
                count[key].push(strs[i]);
            }
        }
        const keys = Object.keys(count);
        for(const k of keys) {
            res.push(count[k]);
        }
        return res;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s1 = set(s)
        s2 = set(t)

        if not len(s) == len(t):
            return False
        
        for c in s1:
            if not c in s2:
                return False

        return True